File: SwiftAppUI/RoamRave/RoamRave/CloseMapView.swift
//
//  CloseMapView.swift
//  RoamRave
//
//  Created by Mari Rosenwald on 4/27/24.
//

import SwiftUI
import MapKit

struct CloseMapView: View {
    var coordinate: CLLocationCoordinate2D


    var body: some View {
        Map(position: .constant(.region(region)))
    }


    private var region: MKCoordinateRegion {
        MKCoordinateRegion(
            center: coordinate,
            span: MKCoordinateSpan(latitudeDelta: 0.001, longitudeDelta: 0.001)
        )
    }
}


#Preview {
    CloseMapView(coordinate: CLLocationCoordinate2D(latitude: 47.6062, longitude: -122.335))
}

File: components/CloseMapView.js
import React from 'react';
import MapView, { Marker } from 'react-native-maps';

const CloseMapView = ({ coordinate }) => {
    return (
        <MapView
            style={{ height: 200, marginVertical: 10 }}
            initialRegion={{
                latitude: coordinate.latitude,
                longitude: coordinate.longitude,
                latitudeDelta: 0.0922,
                longitudeDelta: 0.0421,
            }}
        >
            <Marker coordinate={coordinate} />
        </MapView>
    );
};

export default CloseMapView;
